src/cargo/core/compiler/compilation.rs

```diff
@@ -384,6 +375,48 @@ fn fill_rustc_tool_env(mut cmd: ProcessBuilder, unit: &Unit) -> ProcessBuilder {
     cmd
 }
 
+fn get_sysroot_target_libdir(
+    bcx: &BuildContext<'_, '_>,
+) -> CargoResult<HashMap<CompileKind, PathBuf>> {
+    bcx.all_kinds
+        .iter()
+        .map(|&kind| {
```

Comment:
(nit: styling)

Inside this `map` call, I feel like abusing `let-else` makes things more readable, like:

```rust
            let Some(info) = bcx.target_data.get_info(kind) else {
                ...
                ...
                anyhow::bail!(...);
            };
            Ok((kind, info.sysroot_target_libdir.clone()))
```